source/ui/beamline/AMXYThetaControlMoveButton.cpp

```diff
@@ -0,0 +1,282 @@
+		connect(controlY_, SIGNAL(destroyed()), this, SLOT(onControlYDestroyed()));
+		connect(controlY_, SIGNAL(connected(bool)), this, SLOT(checkStatus()));
+	}
+
+	setText(QString::number(currentStepSize()) % (controlX_ ? controlX_->units() : QString()));
+
+	// enabled / disabled:
+	checkStatus();
+}
+
+void AMXYThetaControlMoveButton::setControlTheta(AMControl *controlR)
+{
+	if(controlR == controlTheta_)
+		return;
+
+	if(contextMenu_) {
+		contextMenu_->deleteLater();
+		contextMenu_ = 0;
+	}
+
+	if(controlTheta_) {
+		disconnect(controlTheta_, 0, this, 0);
+	}
+
+	controlTheta_ = controlR;
+	if(controlX_) {
+		connect(controlTheta_, SIGNAL(destroyed()), this, SLOT(onControlThetaDestroyed()));
+		connect(controlTheta_, SIGNAL(connected(bool)), this, SLOT(checkStatus()));
+	}
+
+	setText(QString::number(currentStepSize()) % (controlX_ ? controlX_->units() : QString()));
+
+	// enabled / disabled:
+	checkStatus();
+}
+
+void AMXYThetaControlMoveButton::onButtonClicked()
+{
+	if(controlX_ && controlY_ && controlTheta_) {
+		controlX_->moveRelative(currentStepSize() * (directionXReversed_ ? -1.0 : 1.0) * sin((controlTheta_->value() + angleOffset_)/57.297));
+		controlY_->moveRelative(currentStepSize() * (directionYReversed_ ? -1.0 : 1.0) * cos((controlTheta_->value() + angleOffset_)/57.297));
+	}
+}
+
+
+void AMXYThetaControlMoveButton::onControlXDestroyed()
+{
+	setControlX(0);
+}
+void AMXYThetaControlMoveButton::onControlYDestroyed()
+{
+	setControlY(0);
+}
+void AMXYThetaControlMoveButton::onControlThetaDestroyed()
+{
+	setControlTheta(0);
+}
+
+void AMXYThetaControlMoveButton::checkStatus()
+{
+	if(controlX_ && controlX_->isConnected() && controlY_ && controlY_->isConnected() && controlTheta_ && controlTheta_->isConnected())
+		setEnabled(true);
+	else
+		setDisabled(true);
+
+}
+
+bool AMXYThetaControlMoveButton::setStepSizeIndex(int index)
+{
+	if(index<0 || index >= stepSizes_.count())
+		return false;
+	if(stepSizeIndex_ != index) {
+		emit stepSizeIndexChanged(stepSizeIndex_ = index);
+	}
+	setText(QString::number(currentStepSize()) % (controlX_ ? controlX_->units() : QString()));
+	return true;
+}
+
+bool AMXYThetaControlMoveButton::setStepSizes(const QList<double> &stepSizes)
+{
+	if(stepSizes.isEmpty())
+		return false;
+
+	// in the future: if we wanted to be slightly more efficient, could update the context menu instead of letting a whole new one be created.
+	if(contextMenu_) {
+		contextMenu_->deleteLater();
+		contextMenu_ = 0;
+	}
+
+	stepSizeIndex_ = 0;
+	stepSizes_ = stepSizes;
+
+	setText(QString::number(currentStepSize()) % (controlX_ ? controlX_->units() : QString()));
+
+	return true;
+}
+
+
+void AMXYThetaControlMoveButton::onCustomContextMenuRequested(const QPoint &point)
+{
+	if(!contextMenu_)
+		contextMenu_ = new AMXYThetaControlMoveButtonContextMenu(this);
+
+	contextMenu_->move(mapToGlobal(point)-QPoint(6,6));
+	contextMenu_->show();
+}
+
+void AMXYThetaControlMoveButton::press(bool down)
+{
+	if(!isEnabled())
```

Comment:
I can answer *most* of these questions with 3 words:  Boots wrote it.

This is a implementation (started as a subclass, but) of his ```AMControlMoveButton```.   As to this point, I don't know.   And I image that the convenience methods that you mentioned above didn't exist at the time.